edumfa/static/components/token/controllers/tokenDetailController.js

```diff
@@ -230,8 +230,9 @@ myApp.controller("tokenDetailController", ['$scope', 'TokenFactory',
     };
 
     $scope.deleteTokenAsk = function() {
-        var tokenType = $scope.token.info.tokenkind;
-        if (tokenType == "hardware"){
+        var tokenKind = $scope.token.info.tokenkind;
+        var typeType = $scope.token.tokentype;
+        if (tokenKind == "hardware" || typeType in $scope.token_ask_delete ){
```

Comment:
this could be problematic:
*token_ask_delete* is just a string, so you do a substring search (e.g "tan" would be in "pushtan").
I think currently we don't have tokentypes that are substrings of others, but we should get this right.